micro_dl/cli/image_inference.py

```diff
@@ -229,33 +261,64 @@ def run_prediction(args, gpu_ids, gpu_mem_frac):
                     np.save(file_name, im_pred, allow_pickle=True)
                 else:
                     raise ValueError('Unsupported file extension')
+
+                # assuming target and predicted images are always 2D for now
+                # Load target
+                meta_idx = aux_utils.get_meta_idx(
+                    frames_meta,
+                    time_idx,
+                    target_channel,
+                    slice_idx,
+                    pos_idx,
+                )
+                file_path = os.path.join(
+                    args.image_dir,
+                    frames_meta.loc[meta_idx, "file_name"],
+                )
+                # get a single row of frame meta data
+                test_frames_meta_row = frames_meta.loc[[meta_idx]]
+                im_target = preprocess_imstack(
+                    frames_metadata=frames_meta,
+                    input_dir=args.image_dir,
+                    depth=1,
+                    time_idx=time_idx,
+                    channel_idx=target_channel,
+                    slice_idx=slice_idx,
+                    pos_idx=pos_idx,
+                )
+                im_target = image_utils.crop2base(im_target)
```

Comment:
this will fail with the current crop2base as crop2base expects a 2D image and crops dim=[0, 1] only. Here after preprocess_imstack, u've to crop along dim=[1, 2] for stack to 2D! u can get the changes from 3D_inference: https://github.com/czbiohub/microDL/blob/b3b8f20a079829b7874238dcbf45c2af8cf37756/micro_dl/utils/image_utils.py#L77, however i haven't tested it yet!